definitions/npm/react-redux_v5.x.x/flow_v0.62.x-/test_connect.js

```diff
@@ -0,0 +1,179 @@
+
+}
+
+function testMapStateToPropsDoesNotNeedProps() {
+  type Props = {passtrough: number, fromStateToProps: string};
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passtrough}</div>;
+    }
+  }
+
+  type State = {a: string}
+  const mapStateToProps = (state: State) => {
+    return {
+      fromStateToProps: state.a
+    }
+  }
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passtrough={123}/>;
+  //$ExpectError component property passtrough not found
+  <Connected />;
+}
+
+function testMapDispatchToProps() {
+  type Props = {
+    passtrough: number,
+    fromMapDispatchToProps: string,
+    fromMapStateToProps: string
+  };
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>
+        {this.props.passtrough}
+        {this.props.fromMapDispatchToProps}
+        {this.props.fromMapStateToProps}
+        </div>;
+    }
+  }
+
+  type State = {a: number}
+  type MapStateToPropsProps = {forMapStateToProps: string}
+  const mapStateToProps = (state: State, props: MapStateToPropsProps) => {
+    return {
+      fromMapStateToProps: state.a
+    }
+  }
+  type MapDispatchToPropsProps = {forMapDispatchToProps: string}
+  const mapDispatchToProps = (dispatch: *, ownProps: MapDispatchToPropsProps) => {
+    return {fromMapDispatchToProps: ownProps.forMapDispatchToProps}
+  }
+  const Connected = connect(mapStateToProps, mapDispatchToProps)(Com);
+  <Connected passtrough={123} forMapStateToProps={'data'} forMapDispatchToProps={'more data'} />;
+  //$ExpectError passtrough missing
+  <Connected forMapStateToProps={'data'} forMapDispatchToProps={'more data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passtrough={123} forMapDispatchToProps={'more data'} />;
+  //$ExpectError forMapDispatchToProps missing
+  <Connected passtrough={123} forMapStateToProps={'data'} />;
+}
+
+function testMapDispatchToPropsWithoutMapStateToProps() {
+  type Props = {
+    passtrough: number,
+    fromMapDispatchToProps: string
+  };
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>
+        {this.props.passtrough}
+        {this.props.fromMapDispatchToProps}
+      </div>;
+    }
+  }
+
+  type MapDispatchToPropsProps = {forMapDispatchToProps: string};
+  const mapDispatchToProps = (dispatch: *, ownProps: MapDispatchToPropsProps) => {
+    return {fromMapDispatchToProps: ownProps.forMapDispatchToProps}
+  }
+  const Connected = connect(null, mapDispatchToProps)(Com);
+  <Connected passtrough={123} forMapStateToProps={'data'} forMapDispatchToProps={'more data'} />;
+  //$ExpectError passtrough missing
+  <Connected forMapStateToProps={'data'} forMapDispatchToProps={'more data'} />;
+  //$ExpectError forMapDispatchToProps missing
+  <Connected passtrough={123} forMapStateToProps={'data'} />;
+}
+
+function testMapDispatchToPropsPassesActionCreators() {
+  type Props = {
+    passtrough: number,
+    dispatch1: () => {},
```

Comment:
This does not match the function you are passing in mapDispatchToProps:  `() => {}`, you probably want `() => void` here since you are not returning an object from `dispatch1`

Reply:
good catch! Seems that Flow  cannot handle the return value of the function. Fortunately it understands what needs to be passed in, i'll add test for that.